src/linear_algebra/basis.lean

```diff
@@ -341,6 +341,33 @@ lemma is_basis.smul_eq_zero [no_zero_divisors R] {c : R} {x : M} :
 
 end
 
+lemma is_basis.smul_of_is_unit {v : ι → M} (hv : is_basis R v)
+  {w : ι → R} (hw : ∀ i : ι, is_unit (w i)) :
+  is_basis R (w • v) :=
+begin
+  obtain ⟨hw₁', hw₁''⟩ := hv,
+  split,
+  { rw linear_independent_iff'' at hw₁' ⊢,
+    intros s g hgs hsum i,
+    obtain ⟨wi, hwi⟩ := hw i,
+    have hw₁ : g i * wi = 0 := hw₁' s (λ i, g i • wi) _ _ i,
+    { rwa [units.mul_left_eq_zero] at hw₁, },
+    { intros i hi,
+      dsimp only,
+      exact (hgs i hi).symm ▸ zero_smul _ _ },
+    { rw [← hsum, finset.sum_congr rfl _],
+      intros, simp only [smul_assoc] } },
```

Comment:
Double checked for invisible red squiggles this time:
```suggestion
  { rw linear_independent_iff'' at hw₁' ⊢,
    intros s g hgs hsum i,
    obtain ⟨wi, hwi⟩ := hw i,
    rw [←wi.mul_left_eq_zero, hwi],
    refine hw₁' s (λ i, g i • w i) (λ i hi, _) _ i,
    { dsimp only,
      exact (hgs i hi).symm ▸ zero_smul _ _ },
    { rw [← hsum, finset.sum_congr rfl _],
      intros, simp only [smul_assoc] } },
```

Reply:
Ah right, I'll take your words for it :)

Reply:
I think there is still an error, at least locally after I pulled :/

Reply:
I think I've fixed it now